src/rgw/rgw_op.cc

```diff
@@ -577,6 +581,43 @@ int rgw_build_object_policies(RGWRados *store, struct req_state *s,
   return ret;
 }
 
+void rgw_add_to_iam_environment(rgw::IAM::Environment& e, const std::string& key, const std::string& val){
+  e.emplace(std::piecewise_construct,
```

Comment:
This will indeed (in C++11) probably skip a temporary, and LGTM-- but, consider also:
m.insert({k, v});
...or even:
m[k] = v;

...sliiiighly less efficient but probably a bit more readable (note a copy will be made in any case since these are const& parameters).